### PR TITLE
Scorecard updates to city in view or closest to center of map

### LINKED ISSUE #112: Dragging map to city should pull up its scorecard
Right now, you can only pull up a city scorecard by using the city toggle or the URL. That's not as interactive. 

To fix this, we would record a bounding box for each city. Then have code like this:

```javascript
var cityBounds = [
    [southWestLat, southWestLng], 
    [northEastLat, northEastLng]
];
var mapViewBounds = map.getBounds();
if (mapViewBounds.intersects(cityBounds)) {
    // Load city data
}
```

We should only change cities when the zoom is close enough.

We probably need some tie breaking mechanism.

### PR DESCRIPTION
**Pull request by tunglinn**
Resolves #112

### CODE REVIEW

**Comment by tunglinn** [comment]
Right now, only the scorecard updates. Should the dropdown also change?

**Comment by Eric-Arellano** [comment]
> Should the dropdown also change?

I think probably? Otherwise you get into a weird UI state where the UI components aren't in sync. What do you think?

**Comment by tunglinn** [comment]
> > Should the dropdown also change?
> 
> I think probably? Otherwise you get into a weird UI state where the UI components aren't in sync. What do you think?

I agree. My concern is how do we differentiate between changes in the dropdown by the user versus by a function. Because if we don't, then changing the dropdown will also 'snap' the map view to the other city.

**Comment by Eric-Arellano** [comment]
>then changing the dropdown will also 'snap' the map view to the other city.

we want that, right? That's the current behavior.

**Comment by tunglinn** [comment]
> we want that, right? That's the current behavior.

Yes, for switching cities via the dropdown. 
But if the dropdown updates when the user is dragging to another city, then the same `setMapToCity()` is called and snaps view to city, which will make it difficult for users to just drag around the map.

File: tests/app/setUpSite.test.js [hunk 1]
```diff
@@ -174,14 +183,11 @@ test.describe("auto-focus city", () => {
...
+
     // Drag map to bring Birmingham into view.
-    await page.locator("#atlanta-ga").hover();
-    await page.mouse.move(-300, 0); // Avoid clicking on Atlanta boundary.
-    await page.mouse.down();
-    await page.mouse.move(500, 0);
-    await page.mouse.up();
-    // Click on Birmingham boundary.
+    await dragMap(page, 200);
+    // Click on Birmingham ;boundary.
```
Comment (Eric-Arellano) [inline tests/app/setUpSite.test.js:190]:
```suggestion
    // Click on Birmingham boundary.
```

**Pull request merged**